charm-slurmd/src/charm.py

```diff
@@ -289,22 +289,24 @@ def _assemble_partition(self):
 
     def _get_set_partition_name(self):
         """Set the partition name."""
-        # Determine if a partition-name config exists, if so
-        # ensure the self._stored.partition_name is consistent with the
-        # supplied config.
+        # Determine if a user-supplied partition-name config exists, if so
+        # ensure the partition_name is consistent with the supplied config.
         # If no partition name has been specified then generate one.
         partition_name = self.config.get("partition-name")
         if partition_name:
             partition_name = partition_name.replace(' ', '-')
-            if partition_name != self._stored.partition_name:
-                self._stored.partition_name = partition_name
+            self._set_partition_name(partition_name)
         else:
-            if not self._stored.partition_name:
-                self._stored.partition_name = f"juju-compute-{random_string()}"
+            if not self.get_partition_name():
+                self._set_partition_name(f"juju-compute-{random_string()}")
+
+    def get_partition_name(self) -> str:
+        """Return the partition_name from the peer relation."""
+        return self._slurmd_peer.partition_name
 
-    def get_partition_name(self):
-        """Return the partition_name."""
-        return self._stored.partition_name
+    def _set_partition_name(self, name: str):
+        """Set the partition_name in the peer relation."""
+        self._slurmd_peer.partition_name = name
```

Comment:
The getter is used in the slurmctld relation as well, that's why it is public. The setter should only be used by leader of the application, and therefore private.